src/serlo-editor/math/editor.tsx

```diff
@@ -271,43 +165,52 @@ export function MathEditor(props: MathEditorProps) {
           </div>
         )}
 
-        {hasError || !isVisualMode ? renderOverlayPortal() : null}
+        {hasError || !isVisualMode ? (
+          <MathEditorOverlayPortal
+            hasError={hasError}
+            isVisualMode={isVisualMode}
+            {...props}
+          />
+        ) : null}
       </>
     )
   }
 
   function renderControlsPortal(children: JSX.Element) {
-    const target =
-      typeof window !== undefined &&
-      document.querySelector('.toolbar-controls-target')
-    if (!target) return null
-
-    return createPortal(children, target)
+    if (!targetRef.current) return null
+    return createPortal(children, targetRef.current)
   }
+}
 
-  function renderOverlayPortal() {
-    return (
-      <div
-        className="fixed bottom-0 z-50 rounded-t-xl bg-editor-primary-100 p-3 shadow-menu"
-        onClick={(e) => e.stopPropagation()} // double/triple clicks close overlay otherwise (#2700)
-      >
-        <div className="flex items-center justify-between">
-          <p className="mr-0.5 mt-1 text-right text-sm font-bold text-gray-600">
-            {hasError ? mathStrings.onlyLatex : mathStrings.latexEditorTitle}
-          </p>
-          <button
-            onClick={props.closeMathEditorOverlay}
-            className="mr-0.5 mt-1 text-sm font-bold text-gray-600 hover:bg-gray-200 hover:text-gray-900 focus:outline-none focus:ring-2 focus:ring-gray-700"
-            aria-label="Close math formula editor"
-            data-qa="plugin-math-close-formula-editor"
-          >
-            <FaIcon icon={faXmark} />
-          </button>
-        </div>
-        {!isVisualMode && (
-          <MathEditorTextArea {...props} defaultValue={state} />
-        )}
+function MathEditorOverlayPortal({
```

Comment:
Why is this called `…OverlayPortal`? 
Also I think I'd prefer to have this in an own file now.